### PR TITLE
fix(mantine): modal spacing

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

This PR includes multiple fixes around modals. Most notably, the overlay opacity, the close button alignment and the modal inner spacing.

<details>
<summary>Simple Modal</summary>
Master
image
Next
image
This branch
image
</details>

<details>
<summary>Modal with tabs</summary>
Master
image
Next
image
This branch
image
</details>

<details>
<summary>Prompt</summary>
Master
image
Next
image
This branch
image
</details>

<details>
<summary>Modal Wizard</summary>
Master
image
Next
image
This branch
image
</details>

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/mantine/src/styles/Modal.module.css
Comment (FelixBlaisThon) [inline packages/mantine/src/styles/Modal.module.css:1]: Were this changes more for clarity ? or was the `& ` useless in this case ?

**Review by FelixBlaisThon — APPROVED** [review]

**Review by toofff — APPROVED** [review]
Many thanks for these changes, I was able to test the demo and found no other problems. :pray:

File: packages/mantine/src/theme/Theme.tsx [hunk 1]
```diff
@@ -122,19 +122,23 @@ export const plasmaTheme: MantineThemeOverride = createTheme({
...
-                body: ModalClasses.body,
+            classNames: ModalClasses,
+        }),
+        ModalRoot: Modal.Root.extend({
+            defaultProps: {
+                padding: 'lg',
             },
         }),
-        ModalOverlay: ModalOverlay.extend({
+        ModalOverlay: Modal.Overlay.extend({
```
[the remark this answers is not in the document]
Comment (gdostie) [reply packages/mantine/src/theme/Theme.tsx:138]: Both work, its just that doing it this way you only need to import the root component. That's how Vivaldi does it in his examples https://mantine.dev/theming/default-props/#default-props-for-compound-components

File: packages/mantine/src/styles/Modal.module.css
Comment (gdostie) [reply packages/mantine/src/styles/Modal.module.css:1]: The `&` are useless

**Review by dmgauthier — APPROVED** [review]

**Pull request merged**